### PR TITLE
[Synapse] Add new cmdlets for Synapse SQL Pool and Synapse Workspace

### PR DESCRIPTION
**Pull request by wonner**
## Description

review PR:https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/772

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@wonner 

Please fix the following issues reported by CI

In generating help
```
Get-AzSynapseSqlActiveDirectoryAdministrator.md Online version in the header of the file is incorrect. 
```

In tests
```
Microsoft.Azure.Commands.Synapse.Test.ScenarioTests.WorkspaceTests.TestSynapseWorkspace [FAIL]
Method not found: 'System.Nullable`1<System.Guid> Microsoft.Azure.Management.Synapse.Models.ManagedIdentity.get_TenantId()'
```

**Comment by wonner** [comment]
Have fixed for the above issues. Please review again. :)

**Comment by idear1203** [comment]
@msJinLei could you please help with a review again?

File: src/Synapse/Synapse/ChangeLog.md [hunk 1]
```diff
@@ -22,6 +22,28 @@
 ## Version 0.4.0
 * Add `-Force` to all Remove cmdlets
 
+* Added support for operation of Synapse SQL Pool Restore Point
```
Comment (msJinLei) [inline src/Synapse/Synapse/ChangeLog.md:25]: The change should be under upcoming release

**Review by msJinLei — APPROVED** [review]

**Review by msJinLei — DISMISSED** [review]

**Comment by idear1203** [comment]
@msJinLei please DO NOT MERGE this PR. We will need to update the SDK to use the official bit. I will let you know when we are ready for a merge.

**Comment by idear1203** [comment]
> CodeFlow for GitHub does not download executable files. If you trust this file and would like to run it, please download it yourself through GitHub.

Please remove these private bits 

---
Refers to: tools/LocalFeed/Microsoft.Azure.Management.Synapse.1.0.0-prevtest.4.1.nupkg:1 in 347ea7f. [](commit_id = 347ea7f102e3f473695ef1ba82a9daf7f68c7eba, deletion_comment = False)

**Review by msJinLei — APPROVED** [review]

**Pull request merged**